sammba/registration/struct.py

```diff
@@ -387,130 +418,85 @@ def anats_to_common(anat_filenames, write_dir, brain_volume,
-        for warp_file, centered_head_file in zip(previous_warp_files,
-                                                 centered_head_files):
-            out_nwarp_cat = nwarp_cat(
-                in_files=[('IDENT', out_tstat_warp_head.outputs.out_file),
-                          warp_file], out_file='iniwarp.nii.gz')
-            out_qwarp = qwarp(
-                in_file=centered_head_file,
-                base_file=out_tstat_warp_head.outputs.out_file,
-                nmi=True,
-                noneg=True,
-                iwarp=True,
-                weight=out_mask_tool.outputs.out_file,
-                iniwarp=[out_nwarp_cat.outputs.out_file],
-                inilev=nonlinear_levels[0],
-                maxlev=nonlinear_levels[1],
-                out_file=fname_presuffix(centered_head_file,
-                                         suffix='_warped2'))
-            warp_files.append(out_qwarp.outputs.source_warp)
-            warped_files.append(out_qwarp.outputs.warped_source)
-
-        out_tcat = tcat(in_files=warped_files,
-                        out_file='warped_2iters_heads.nii.gz')
-        out_tstat_warp_head = tstat(in_file=out_tcat.outputs.out_file,
-                                    outputtype='NIFTI_GZ')
 
-    ###########################################################################
-    # Using previous files and concatenated transforms can be exploited to
-    # avoid building up reslice errors.
-    # Warp with mini-patch
-    # In this particular case, minpatch=75 corresponds to a level of 4
-    if len(nonlinear_levels) > 2:
-        if nonlinear_minimal_patch is None:
-            nonlinear_minimal_patch = 75
-
-        for n_iter, inilev in enumerate(nonlinear_levels[2:]):
-            previous_warp_files = warp_files
-            warped_files = []
-            warp_files = []
-            for warp_file, centered_head_file in zip(previous_warp_files,
-                                                     centered_head_files):
-                suffixed_file = fname_presuffix(
-                    centered_head_file,
-                    suffix='_warped{}'.format(n_iter + 3))
-                if n_iter == len(nonlinear_levels):
-                    out_file = os.path.join(write_dir,
-                                            os.path.basename(suffixed_file))
-                else:
-                    out_file = suffixed_file
+        out_file = fname_presuffix(centered_head_file, suffix='_warped{}'.format(n_iter))
+    
+        for warp_file, centered_head_file in zip(previous_warp_files, 
+                                                 centered_head_files):
 
+            if n_iter == 0:
+                out_qwarp = qwarp(
+                    in_file=centered_head_file,
+                    base_file=out_tstat_allineated_head.outputs.out_file,
+                    noneg=True,
+                    iwarp=True,
+                    weight=nonlinear_weight,
+                    iniwarp=warp_file,
```

Comment:
in nipype Qwarp interface, `iniwarp` should be a list of existing files. Did you check that when you run the function you din't get a trait type error ?

Reply:
actually I checked and it throws an error
>TraitError: The 'iniwarp' trait of a QwarpInputSpec instance must be a list of items which are an existing file name, but a value of u'/tmp/pauline/template11/MSME_MIRCen_RatBrain_atlas_bf5_0_Unifized_for_brain_extraction_resample_shr_affine_catenated.aff12.1D' <type 'unicode'> was specified.

So this must be corrected

Reply:
Corrected. Dunno how I never got an error.